cartridges/int_adyen_controllers/cartridge/controllers/Adyen.js

```diff
@@ -63,26 +63,38 @@ function redirect(order, redirectUrl) {
  * Show confirmation after return from Adyen
  */
 function showConfirmation() {
-	var payLoad = request.httpParameterMap.payload.value;
-	//redirect to payment/details
-	var adyenCheckout = require('*/cartridge/scripts/adyenCheckout');
-	var requestObject = {};
-	requestObject['details'] = {};
-	requestObject.details['payload'] = payLoad;
-	var result = adyenCheckout.doPaymentDetailsCall(requestObject);
-	var orderNumber = result.merchantReference;
+    var orderNumber = session.privacy.orderNo;
     var order = OrderMgr.getOrder(orderNumber);
 
     var paymentInstruments = order.getPaymentInstruments("Adyen");
     var adyenPaymentInstrument;
+    var paymentData;
+
     var instrumentsIter = paymentInstruments.iterator();
     while (instrumentsIter.hasNext()) {
         adyenPaymentInstrument = instrumentsIter.next();
-        Transaction.wrap(function () {
-            adyenPaymentInstrument.custom.adyenPaymentData = null;
-        });
+        paymentData = adyenPaymentInstrument.custom.adyenPaymentData;
     }
 
+    //details is either redirectResult or payload
+    var details;
+    if(request.httpParameterMap.redirectResult.value != null){
```

Comment:
will this work if redirectResult is undefined or emtpy because you do a .value to this ?

Reply:
Yes, then it will go into the else if